services/wfm/src/main/java/org/openkilda/wfm/topology/event/OFELinkBolt.java

```diff
@@ -483,25 +511,45 @@ private void handleIslEvent(Tuple tuple, IslInfoData discoveredIsl) {
     //          - services/src/topology .. service/impl/IslServiceImpl .. service/IslService
     //          - services/src/topology .. messaging/kafka/KafkaMessageConsumer
     //          - services/src/pce .. NetworkCache .. FlowCache ..
-    private void sendDiscoveryFailed(String switchId, String portId, Tuple tuple, String correlationId) throws IOException {
+    private void sendDiscoveryFailed(String switchId, int portId, Tuple tuple, String correlationId)
+            throws IOException {
         String discoFail = OFEMessageUtils.createIslFail(switchId, portId, correlationId);
-//        Values dataVal = new Values(PAYLOAD, discoFail, switchId, portId, OFEMessageUtils.LINK_DOWN);
-//        collector.emit(topoEngTopic, tuple, dataVal);
+        //        Values dataVal = new Values(PAYLOAD, discoFail, switchId, portId, OFEMessageUtils.LINK_DOWN);
+        //        collector.emit(topoEngTopic, tuple, dataVal);
         collector.emit(topoEngTopic, tuple, new Values(PAYLOAD, discoFail));
         discovery.handleFailed(switchId, portId);
         logger.warn("LINK: Send ISL discovery failure message={}", discoFail);
     }
 
     private boolean isPortUpOrCached(String state) {
-        return OFEMessageUtils.PORT_UP.equals(state) || OFEMessageUtils.PORT_ADD.equals(state) ||
-                PortChangeType.CACHED.getType().equals(state);
+        return OFEMessageUtils.PORT_UP.equals(state) || OFEMessageUtils.PORT_ADD.equals(state)
+                || PortChangeType.CACHED.getType().equals(state);
     }
 
     private void enableDumpRequestTimer() {
         long expireDelay = (int) (dumpRequestTimeout * 1000);
         dumpRequestTimer = new Timer(expireDelay, expireDelay, TimeUnit.MILLISECONDS);
     }
 
+    private void handleMovedIsl(Tuple tuple, String srcSwitch, int srcPort, String dstSwitch, int dstPort) {
+        NetworkEndpoint dstEndpoint = discovery.getLinkDestination(srcSwitch, srcPort);
+        logger.info("Link is moved from {}_{} - {}_{} to endpoint {}_{}", srcSwitch, srcPort,
+                dstEndpoint.getSwitchDpId(), dstEndpoint.getPortId(), dstSwitch, dstPort);
+
+        PathNode srcNode = new PathNode(srcSwitch, srcPort, 0);
+        PathNode dstNode = new PathNode(dstEndpoint.getSwitchDpId(), dstEndpoint.getPortId(), 1);
+        IslInfoData infoData = new IslInfoData(Lists.newArrayList(srcNode, dstNode), IslChangeType.MOVED);
+        InfoMessage message = new InfoMessage(infoData, System.currentTimeMillis(), UUID.randomUUID().toString());
```

Comment:
The recent code has correlationId as an argument in handleIslEvent, so you can/should reuse it.